File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cs
public class Solution 
{
    public int solution(int num) 
    {
        long value = num;
        int count = 0;

        while (value != 1 && count < 500)
        {
            if (value % 2 == 0)
            {
                value /= 2;
            }
            else
            {
                value = value * 3 + 1;
            }

            count++;
        }

        return value == 1 ? count : -1;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) 
{
    long long n = num ;
    int count = 0;
    while (n != 1) // num이 1이 아닐 때 반복
    {
        if (n % 2 == 0) // 짝수
        {
            n /= 2;
        }
        else // 홀수
        {
            n = n * 3 + 1;
        }
        count++;
    }
    
    return (count < 500) ? count : -1;
}